custom_components/tuya_local/devices/agl_ultracontato.yaml

```diff
@@ -0,0 +1,84 @@
+name: AGL ultracontato r2
+primary_entity:
+  entity: cover
+  class: door
+  dps:
+    - id: 1
+      name: control
+      type: boolean
+      mapping:
+        - dps_val: true
+          value: "open"
+        - dps_val: false
+          value: "close"
+    - id: 101
+      name: open
+      type: boolean
+secondary_entities:
+  - entity: select
+    name: Switch type
+    category: config
+    icon: "mdi:electric-switch-closed"
+    dps:
+      - id: 108
+        type: string
+        name: option
+        mapping:
+          - dps_val: "NA"
+            value: Normally Open
+          - dps_val: "NF"
+            value: Normally Closed
+  - entity: select
+    category: config
+    name: Pulse type
+    icon: "mdi:pulse"
+    dps:
+      - id: 109
+        name: option
+        type: string
+        mapping:
+          - dps_val: "Pulsado"
+            value: Pulsed
+          - dps_val: "Retencao"
+            value: Retencion
```

Comment:
I think name should be Output type, since pulsed is just one of the options.  I think the first two are what is commonly referred to as Pulse and Level in electrical engineering terms, but I'm not sure where the ONOFF option fits in.  Maybe the Installation manual has enough details to work out what these really mean.  Retention would be the literal translation of Rentenção in English (Retencion looks more like a Spanish spelling), but doesn't make much sense in this context.